### PR TITLE
Change r/runbook step config attribute type to a JSON string

### PR DESCRIPTION
**Pull request by lafentres**
## Description

This PR changes the type of the steps config attribute to a JSON string to fix various error preventing some runbook steps from being imported or created successfully.

All valid options for the steps config attribute will be documented in a separate PR.

## Testing plan

The goal of this test plan is to check that that the updates and new validations work as expected, get updated when they should, etc. 

Pre-requisites:
- You'll need Go 1.16 and the latest Terraform installed.
- You'll need an instance of FireHydrant running locally or an organization in production to test against.
- You'll need a bot token for the organization you plan to test against.

#### Setup
1. Create a new directory called `providers` somewhere you can easily access (but not inside the same folder as your Terraform config).
1. Build the provider by checking out this branch and running `make build`
1. Move the executable created to the `providers` directory you created earlier.
   ```
   mv terraform-provider-firehydrant /PATH/TO/YOUR/DIRECTORY/providers
   ```
1. Add the following to your ~/.terraformrc file:
   ```
   provider_installation {

     # Use /PATH/TO/YOUR/DIRECTORY/providers/terraform-provider-firehydrant
     # as an overridden package directory for the firehydrant/firehydrant provider. 
     # This disables the version and checksum verifications for this provider and 
     # forces Terraform to look for the null provider plugin in the given directory.
     dev_overrides {
       "firehydrant/firehydrant" = "/PATH/TO/YOUR/DIRECTORY/providers"
     }

     # For all other providers, install them directly from their origin provider
     # registries as normal. If you omit this, Terraform will _only_ use
     # the dev_overrides block, and so no other providers will be available.
     direct {}
   }
   ```
1. Create another new directory (all Terraform commands will be run in this directory)
1.  [Download and save this file](https://github.com/firehydrant/terraform-provider-firehydrant/files/9135473/terraform-runbook-steps-config.txt)as `main.tf` in your new directory. You'll be modifying it in the steps below. 
1. In you FH org, create a runbook. Then take the id of your runbook and replace the placeholder in the config with it. 
1. Run `terraform init`.

#### Provision resources and data sources:
1. Run `terraform apply`. When prompted for an API key, provide your bot token. This should succeed and if you check in the UI, you should see the various resources you just created. 
1. Run `terraform plan`. There should be no changes.

#### Make sure updating things still works
1. Update your config by changing something in a step config. Try removing steps, updating fields, etc.
1. Run `terraform apply`. This should succeed and you should see your updates reflected in the UI and statefile.
1. Run `terraform plan`. You should see no changes.

#### Make sure destroy still works
1. Run `terraform destroy`. This should succeed. 

#### Make sure importing works:
Previously importing didn't work consistently, especially with our runbook templates.
1. Run `terraform apply`. This should succeed and you should see your updates reflected in the UI and statefile.
1. Run `terraform plan`. You should see no changes.
1. Create a new runbook using out started template. Copy the ID
1. In your main.tf file, uncomment the imported_runbook resource
1. Run `terraform import firehydrant_runbook.imported_runbook YOUR_RUNBOOK_ID`. This should succeed and you should see accurate information in the statefile for this runbook.

## Related links

- [API documentation](https://developers.firehydrant.io/docs/api/6d750fbd2d88c-create-a-runbook)

## PR readiness 

- [x] Relevant documentation has been updated if this PR adds or updates attributes, resources, or data sources.
- [x] An entry has been added to the changelog, if necessary.

### CODE REVIEW

File: provider/runbook_resource_test.go [hunk 1]
```diff
@@ -333,98 +391,134 @@ func testAccCheckRunbookResourceDestroy() resource.TestCheckFunc {
...
+
+    config = jsonencode({
+      channel_name_format = "-inc-{{ number }}"
+    })
+  }
 }`, rName)
 }
 
-func testAccRunbookResourceConfig_required_repeats_not_set(rName string) string {
+func testAccRunbookResourceConfig_stepsRequiredRepeatsNotSet(rName string) string {
```
Comment (nearbycoder) [inline provider/runbook_resource_test.go:481]: Ah so thats the casing convention, I was really thrown off by why have underscore and camelcase in places

File: provider/runbook_resource_test.go [hunk 2]
```diff
@@ -286,19 +325,38 @@ func testAccCheckRunbookResourceExistsWithAttributes_update(resourceName string)
...
+				config, err = json.Marshal(step.Config)
+				if err != nil {
+					return fmt.Errorf("Unexpected error converting runbook step config to JSON: %v", err)
+				}
+			}
+			if runbookResource.Primary.Attributes[key+".config"] != string(config) {
+				return fmt.Errorf("Unexpected runbook step config. Expected %s, got: %s", step.Config, runbookResource.Primary.Attributes[key+".config"])
+			}
+
+			if runbookResource.Primary.Attributes[key+".repeats"] != fmt.Sprintf("%t", step.Repeats) {
```
Comment (nearbycoder) [inline provider/runbook_resource_test.go:353]: Oh nice assuming this will give a more detailed error around which step key actually failed.

**Review by nearbycoder — APPROVED** [review]
Pulled this down and went through the steps in the PR description and verified everything was working correctly 
image

File: provider/runbook_resource_test.go (hunk 1 above)
Comment (lafentres) [reply provider/runbook_resource_test.go:481]: i think it is but honestly, i've seen things be a lil different across multiple go repos, which makes things very confusing. this is what they seem to do most of the time in the aws and google providers, which is why i try to do it this way. but we could also kinda do whatever we want i think, as long as it's consistent

File: provider/runbook_resource_test.go (hunk 2 above)
Comment (lafentres) [reply provider/runbook_resource_test.go:353]: that's what i'm hoping 🤞🏼 i tried to get it to fail just so i could confirm that it really does what i think it should but i couldn't get it to work.

**Pull request merged**